src/Client/ClientBase.cpp

```diff
@@ -2294,6 +2297,7 @@ void ClientBase::init(int argc, char ** argv)
         ("interactive", "Process queries-file or --query query and start interactive mode")
         ("pager", po::value<std::string>(), "Pipe all output into this command (less or similar)")
         ("max_memory_usage_in_client", po::value<int>(), "Set memory limit in client/local server")
+        ("send-cancel-after", po::value<int>(), "Send cancel packet after some milliseconds, should be used only for testing")
```

Comment:
Why not simply `timeout -sINT` (`clickhouse_client_timeout`) ?

Reply:
Because clickhouse-client (debug build) takes time to run. I'll use timeout direct and see if it's flaky or not.

Reply:
>Because clickhouse-client (debug build) takes time to run

Indeed, and sometimes server can hung.
So I'm using the following to avoid this:
 https://github.com/ClickHouse/ClickHouse/blob/9fb1d92ff54518f6d6d16b8f3c6c0bf1ed9c0b0a/tests/queries/0_stateless/02290_client_insert_cancel.sh#L7-L18

Reply:
But how `--kill-after=DURATION` will help?

You have `timeout -k 5 -s INT 5 ${CLICKHOUSE_CLIENT} --query_id="${QUERY_ID}" --query='select (select max(number) from system.numbers) + 1;' `, this query will never be finished w/o cancelling, so `timeout` will send INT and exit code of timeout in this case will be 124, because you do not pass `--preserve-status`.

Reply:
Yeah, I've modified the test to check `system.processes` directly instead of relying on return status.